runtime/protocol/http-client/common/src/aws/smithy/kotlin/runtime/http/interceptors/FlexibleChecksumsResponseInterceptor.kt

```diff
@@ -32,74 +31,72 @@ internal val CHECKSUM_HEADER_VALIDATION_PRIORITY_LIST: List<String> = listOf(
 )
 
 /**
- * Validate a response's checksum.
+ * Handles response checksums.
  *
  * If it's a streaming response, it wraps the response in a hashing body, calculating the checksum as the response is
  * streamed to the user. The checksum is validated after the user has consumed the entire body using a checksum validating body.
  * Otherwise, the checksum if calculated all at once.
  *
  * Users can check which checksum was validated by referencing the `ResponseChecksumValidated` execution context variable.
  *
- * @param responseValidationRequired Flag indicating if the checksum validation is mandatory.
+ * @param responseValidationRequired Model sourced flag indicating if the checksum validation is mandatory.
  * @param responseChecksumValidation Configuration option that determines when checksum validation should be done.
  */
 @InternalApi
-public class FlexibleChecksumsResponseInterceptor(
+public class FlexibleChecksumsResponseInterceptor<I>(
     private val responseValidationRequired: Boolean,
     private val responseChecksumValidation: HttpChecksumConfigOption?,
+    private val serviceUsesCompositeChecksums: Boolean,
```

Comment:
question: Why do we need this extra flag to tell if the operation **might** send back a composite checksum? Why can't we just rely on `serviceChecksumValue.isCompositeChecksum()`?